independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Injection.java

```diff
@@ -296,16 +310,78 @@ private static List<AnnotationInstance> getAllInjectionPoints(BeanDeployment bea
                 }
             }
         }
+
         for (MethodInfo method : beanClass.methods()) {
             if (skipConstructors && method.name().equals(Methods.INIT)) {
                 continue;
             }
-            AnnotationInstance inject = beanDeployment.getAnnotation(method, name);
+
+            MethodOverrideKey key = new MethodOverrideKey(method);
+            if (seenMethods.contains(key)) {
+                continue;
+            }
+            seenMethods.add(key);
+
+            AnnotationInstance inject = beanDeployment.getAnnotation(method, annotationName);
             if (inject != null) {
-                injectAnnotations.add(inject);
+                if (Modifier.isStatic(method.flags())) {
+                    LOGGER.warn("An initializer method must be non-static - ignoring: "
+                            + method.declaringClass().name() + "#"
+                            + method.name() + "()");
+                } else {
+                    injectAnnotations.add(inject);
+                }
             }
         }
+
         return injectAnnotations;
     }
 
+    // ---
+
+    // this is not entirely precise (according to the JLS), but it's better than Methods.MethodKey
```

Comment:
In which sense is it better? Shouldn't we rather improve the `Methods.MethodKey` then?

Reply:
I was thinking about improving `MethodKey`, but I looked at its usages and wasn't entirely sure it wouldn't break everyone. I'll take a look again.

Reply:
Done -- improved `Methods.MethodKey` and `Methods.isOverridden()`, removed `Injection.MethodOverrideKey`.